install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml

```diff
@@ -239,7 +239,11 @@ spec:
 {{- with .Values.extraEnv }}
 {{ toYaml . | trim | indent 8 }}
 {{- end }}
+{{- if .Values.image.useDigest }}
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}@{{ .Values.image.digest }}
+{{- else }}
```

Comment:
Just wondering, could we eliminate `useDigest` and simply rely on `len(digest) != 0`? I think `empty` would work: https://helm.sh/docs/chart_template_guide/function_list/#empty

My thinking is that adding 1 new variable is better than 2.

Reply:
right but there might be users that don't want to use the SHA for some reason, that's why there are 2 variables instead of 1.

Reply:
I'm thinking we could just rely `digest` itself being set or not, rather than have a separate `useDigest` boolean. For example with:
```yaml
{{- if .Values.image.digest }}
        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}@{{ .Values.image.digest }}
{{- else }}
```
If `digest` is `""` as in the `README` examples then it won't enter the `if` block.

Reply:
@Skymirrh the digests will be part of the helm charts but using them will be optional. For example our CI will not use the image digests, even tho they are defined, so the digests will need to be set false when setting up our CI (once we branch off to 1.10)

Reply:
I agree, but even then, doesn't it make sense to just programmatically remove `digest` or set `digest: ""` to clear digest usage in CI, rather than set `useDigest: false`? What I'm trying to say is that since we already have a manual action needed for setting up values for CI anyway, why not just remove complexity by having a single value instead of a separate flag?

Reply:
I think I got it now, yes that makes sense. However, the PRs on the stable branches will need to have the `useDigest` because we want to have the digests available but they should not be mandatory to users, as they it would be a breaking change.

Reply:
I'm a bit confused: could you detail in which scenario it is possible to have `digest: "{someSha}"` but at the same time not wanting to use it?

I was under the assumption that we could limit ourselves to these two scenarios:
- `digest` not present or `digest: ""` > current behavior (no breaking change)
- `digest` present > new behavior

Reply:
@Skymirrh in stable branches, not master we don't want to introduce this new behavior.

Reply:
Ah, so if I understood correctly, in stable branches, we will have:
```yaml
image:
    digest: "{someSha}"
    useDigest: false
```
Because we want users to be able to easily switch `useDigest` to `true` and use some predefined digests. Correct?

Reply:
@Skymirrh correct.

Reply:
OK, works for me! Then a small nitpick proposal: instead of
```helm
{{- if .Values.image.digest }}
        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}@{{ .Values.image.digest }}
{{- else }}
        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
{{- end }}
```
we could have
```helm
        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}{{- if .Values.image.useDigest }}@{{ .Values.image.digest }}{{- end }}
```
I tend to use the latter construct when it's a simple flag switch added to an existing block, rather than a wholly different block.

@christarazi feel free to resolve when/if you are satisfied :)

Reply:
@Skymirrh I gave it a little more thought over the weekend. I think it's better to leave the `useDigest` mostly for the cilium-operator where we have 3 different image versions. Picking the right image to run in the right cluster is hidden from the user, i.e., the image chosen depends on the options users set. Since this behavior is hidden, users would need to be aware which digest (`awsDigest`, `azureDigest`, etc) they would need to disable. Let me know what do you think.

Reply:
Makes sense. I wonder to what extend we should document this.